lib/global-styles.php

```diff
@@ -614,25 +614,30 @@ function gutenberg_experimental_global_styles_get_editor_features( $config ) {
  * @return array New block editor settings
  */
 function gutenberg_experimental_global_styles_settings( $settings ) {
+	if ( ! function_exists( 'get_current_screen' ) ) {
+		return false;
+	}
+	$screen = get_current_screen();
 
-	if ( gutenberg_experimental_global_styles_has_theme_json_support() ) {
+	if ( ! empty( $screen ) &&
+		gutenberg_is_edit_site_page( $screen->id ) &&
+		gutenberg_experimental_global_styles_has_theme_json_support() ) {
 		$settings['__experimentalGlobalStylesUserEntityId'] = gutenberg_experimental_global_styles_get_user_cpt_id();
-
-		$global_styles = gutenberg_experimental_global_styles_merge_trees(
+		$settings['__experimentalGlobalStylesContexts']     = gutenberg_experimental_global_styles_get_block_data();
+		$settings['__experimentalGlobalStylesBaseStyles']   = gutenberg_experimental_global_styles_merge_trees(
 			gutenberg_experimental_global_styles_get_core(),
 			gutenberg_experimental_global_styles_get_theme()
 		);
-
-		$settings['__experimentalGlobalStylesBase'] = $global_styles;
+	} else {
+		// We are in a block editor context, as this function
+		// is hooked to the block_editor_settings filter.
+		//
+		// Add the styles for the editor via the settings
+		// so they get processed as if they were added via add_editor_styles:
+		// they will get the editor wrapper class.
+		$settings['styles'][] = array( 'css' => gutenberg_experimental_global_styles_get_stylesheet() );
 	}
 
-	// Add the styles for the editor via the settings
-	// so they get processed as if they were added via add_editor_styles:
-	// they will get the editor wrapper class.
-	$merged               = gutenberg_experimental_global_styles_get_merged_origins();
-	$stylesheet           = gutenberg_experimental_global_styles_resolver( $merged );
-	$settings['styles'][] = array( 'css' => $stylesheet );
-
 	$settings['__experimentalFeatures'] = gutenberg_experimental_global_styles_get_editor_features( $merged );
```

Comment:
We are removing the definition of $merged, but still using the variable here. There are notices regarding the usage of an undefined variable.

Reply:
Fixed at https://github.com/WordPress/gutenberg/pull/24250/commits/7f8961dd4b0a30b5bf91022bb6658c9f38676dfe